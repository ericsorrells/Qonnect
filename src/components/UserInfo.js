// ========================================================================================
import React from 'react';
import PropTypes from 'prop-types';
import { Router, Route, Switch, Link, NavLink } from 'react-router-dom';
import moment from 'moment';
// ========================================================================================

const UserInfo = ({
  displayName,
  firstName,
  email,
  photoURL,
  lastName,
  city,
  state,
  birthDate,
  createdAt,
  description,
  urlParam,
  userId
 }) => {
  photoURL = photoURL ? photoURL : "/images/user-silhouette.png"
  return (
    <div className="user-info">
      <div className='user-info__photoURL'>
        <img className="user-info__silhouete" src={photoURL} />
      </div>
      <h1>{firstName} {lastName}</h1>
      <div className="user-info__personal-stats">
        <div className="user-info__item">
          <i className="fas fa-globe"></i>
          {city}, {state}
        </div>
        <div className="user-info__item">
          <i className="fas fa-birthday-cake"></i>
          { moment().diff(birthDate, 'years') } Years Old
        </div>
        <div className="user-info__item">
          <i className="fas fa-sign-in-alt"></i>
          {moment(createdAt).format("MMMM Do YYYY")}
        </div>
      </div>
      <div className="user-info__description">
        {description}
      </div>
      {
        urlParam === userId &&
          <button className='button__purple'>
            <Link to='/update-profile'>Update</Link>
          </button>
      }
    </div>
  )
}

UserInfo.propTypes = {
  user: PropTypes.object
}

export default UserInfo;