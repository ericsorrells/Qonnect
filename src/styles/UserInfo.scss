.user-info {
  font-size: .9rem;
  width: 90%;
  position: relative;
  top:-150px;
}

.user-info__photoURL {
  text-align: center;
}

.user-info__silhouete{
  max-width: 12rem;
  border: 1px solid black;
  background-color: white;
  padding: 30px 10px 0 10px;
}

.user-info__personal-stats,
.user-info__description {
  color: black;
}

.user-info__description {
  margin-bottom: 50px;
}

.user-info__personal-stats,
.user-info h1 {
  padding-bottom: 10px;
  border-bottom: 1px solid black;
  margin-bottom: 20px;
}

.user-info__item {
  display: flex;
  align-items: center;
  margin-bottom: 1.5rem;
  margin-right: 10px;
}

.fa-globe,
.fa-birthday-cake,
.fa-sign-in-alt {
  color: $main-color;
  font-size: 1.25rem;
  margin-right: 1rem;
}
