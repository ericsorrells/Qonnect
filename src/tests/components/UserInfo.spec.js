// ========================================================================================
import React from 'react';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import mockStoreSetup from '../../utils/mockStoreHelpers';
import 'jest-enzyme';
// ========================================================================================
import UserInfo from '../../components/UserInfo';
// ========================================================================================
import { basicEvent } from '../stubs/event';
import { basicProfile } from '../stubs/profile'
// ========================================================================================

describe('<UserInfo/>', () => {
  const component = shallow(
    <UserInfo
    displayName={'coolUser'}
    firstName={basicProfile.firstName}
    email={'johndoe@email.com'}
    photoURL={basicProfile.photoURL}
    lastName={basicProfile.lastName}
    city={basicProfile.city}
    state={basicProfile.state}
    birthDate={basicProfile.birthDate}
    createdAt={basicProfile.createdAt}
    description={basicProfile.description}
    urlParam={'abc123'}
    userId={'abc123'}
    />
  )

  it('renders correctly', () => {
    expect(component).toMatchSnapshot();
  });

  describe('Update button', () => {
    it('allows correct user to update their info', () => {
      expect(component.find('button')).toExist();
    });

    it('prevents incorrect user from updating anothers info', () => {
      component.setProps({ userId: 'xyz789' });
      expect(component.find('button')).not.toExist();
    });
  });
});
