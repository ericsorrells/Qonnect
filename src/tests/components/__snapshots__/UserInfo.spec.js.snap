// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserInfo/> renders correctly 1`] = `
<div
  className="user-info"
>
  <div
    className="user-info__photoURL"
  >
    <img
      className="user-info__silhouete"
      src="https://ih0.redbubble.net/image.14393202.3421/flat,800x800,070,f.u1.jpg"
    />
  </div>
  <h1>
    Rachel
     
    Deckard
  </h1>
  <div
    className="user-info__personal-stats"
  >
    <div
      className="user-info__item"
    >
      <i
        className="fas fa-globe"
      />
      The Sprawl!!!
      , 
      Ga
    </div>
    <div
      className="user-info__item"
    >
      <i
        className="fas fa-birthday-cake"
      />
      41
       Years Old
    </div>
    <div
      className="user-info__item"
    >
      <i
        className="fas fa-sign-in-alt"
      />
      May 11th 2018
    </div>
  </div>
  <div
    className="user-info__description"
  >
    I am a replicant! I think. I know...!
  </div>
  <button
    className="button__purple"
  >
    <Link
      replace={false}
      to="/update-profile"
    >
      Update
    </Link>
  </button>
</div>
`;
